### PR TITLE
VBLOCKS-1211 End-to-end tests and app under "test" folder

### PR DESCRIPTION
**Pull request by bobiechen-twilio**
## Submission Checklist

 - [ ] Updated the `CHANGELOG.md` to reflect any **feature**, **bug fixes**, or **known issues** made in the source code
 - [ ] Tested code changes and observed expected behavior in the example app
 - [ ] Performed a visual inspection of the `Files changed` tab prior to submitting the pull request for review to ensure proper usage of the style guide

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

## Description

Moving the e2e test host app under the `/test` folder

## Breakdown

- Move and rename `test-application` to `test/app`
- Update path references

## Validation

- CircleCI

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -48,7 +48,7 @@ commands:
       - run:
           name: Install cocoapods if cache does not exist
           command: |
-            cd test-application/ios && pod install && cd ../..
+            cd test/app/ios && pod install && cd ../..
```
Comment (mhuynh5757) [inline .circleci/config.yml:51]: Do we need to `cd ../../..` since we're nested one more layer?

**Comment by mhuynh5757** [comment]
We probably should have discussed this, but since we're not moving unit tests does this PR still make sense? It seems like we're going from `test-app` to `test/app`

File: package.json [hunk 2]
```diff
@@ -14,15 +14,15 @@
     "ios",
     "cpp",
     "twilio-voice-react-native.podspec",
-    "!lib/typescript/test-application",
+    "!lib/typescript/app",
```
Comment (mhuynh5757) [inline package.json:17]:
```suggestion
    "!lib/typescript/test/app",
```

File: test/app/.detoxrc.json [hunk 3]
```diff
@@ -1,14 +1,25 @@
...
         }
+    },
+    "configurations": {
+        "ios.release": {
+            "app": "TwilioVoiceReactNativeExample.ios",
+            "device": "simulator"
+        }
     }
-}
+}
```
Comment (mhuynh5757) [inline test/app/.detoxrc.json:25]:
```suggestion
}

```

File: test/app/README.md [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 - [React Native CLI](https://reactnative.dev/docs/environment-setup)
 - [Yarn](https://yarnpkg.com/getting-started)
-- Token - Create a file, `e2e-tests-token.ts` under `test-application/src` directory with the following content.
+- Token - Create a file, `e2e-tests-token.ts` under `app/src` directory with the following content.
```
Comment (mhuynh5757) [inline test/app/README.md:7]:
```suggestion
- Token - Create a file, `e2e-tests-token.ts` under `test/app/src` directory with the following content.
```

**Comment by charliesantos** [comment]
@bobiechen-twilio do we know why ios e2e test is failing?

File: .circleci/config.yml [hunk 5]
```diff
@@ -64,15 +64,15 @@ commands:
...
     steps:
       - run:
           working_directory: .
           name: Clear detox cache
           command: |
-            cd test-application
+            cd test/app
             detox clean-framework-cache
             detox build-framework-cache
             cd ../
```
Comment (charliesantos) [inline .circleci/config.yml:78]: For all the `cd ../` commands here and below, do we need to go one more level up?

File: test/app/README.md [hunk 6]
```diff
@@ -27,9 +27,9 @@
   ```sh
   yarn run build:constants
   ```
-5. Switch to the test-application directory
+5. Switch to the app directory
   ```sh
-  cd test-application
+  cd app
```
Comment (charliesantos) [inline test/app/README.md:32]: > cd test/app

File: test/app/README.md [hunk 7]
```diff
@@ -43,9 +43,9 @@
 
 ## Running the app
 
-Switch to the test-application directory
+Switch to the app directory
 ```sh
-cd test-application
+cd app
```
Comment (charliesantos) [inline test/app/README.md:48]: >cd test/app

**Review by charliesantos — CHANGES_REQUESTED** [review]

**Review by charliesantos — APPROVED** [review]

**Pull request merged**